File: solutions/60-Single-Number-III.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* singleNumber(int* nums, int numsSize, int* returnSize) {
    *returnSize=2;
    int *arr=calloc(2,sizeof(int));
    long int ab = 0, right_most;

        for(int i=0;i < numsSize; i++ ){
            ab^=nums[i];
        }
        right_most = ab & -ab;

        for(int j = 0; j<numsSize; j++){
            if (nums[j]&right_most){
                arr[0]^=nums[j];
            }
            else{
                arr[1]^=nums[j];
            }
        }
    return arr;  
}

File: solutions/60-Single-Number-III.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        ab = 0
        for i in nums:
            ab^=i
        
        right_most = ab & -ab

        a = 0
        b = 0
        for j in nums:
            if j & right_most:
                a^=j
            else:
                b^=j
        return [a,b] 